packages/istanbul-reports/test/fixtures/specs/100-line-100-branch.json

```diff
@@ -1,6 +1,6 @@
 {
   "title": "handles 100% branch and line coverage",
-  "textReportExpected": "----------|----------|----------|----------|----------|-------------------|\nFile      |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |\n----------|----------|----------|----------|----------|-------------------|\n\u001b[32;1mAll files\u001b[0m |\u001b[32;1m      100\u001b[0m |\u001b[32;1m      100\u001b[0m |\u001b[32;1m      100\u001b[0m |\u001b[32;1m      100\u001b[0m |\u001b[33;1m                  \u001b[0m |\n\u001b[32;1m index.js\u001b[0m |\u001b[32;1m      100\u001b[0m |\u001b[32;1m      100\u001b[0m |\u001b[32;1m      100\u001b[0m |\u001b[32;1m      100\u001b[0m |\u001b[33;1m                  \u001b[0m |\n----------|----------|----------|----------|----------|-------------------|\n",
+  "textReportExpected": "---------|-------|--------|-------|-------|-----------------\nFile     |% Stmts|% Branch|% Funcs|% Lines|Uncovered Line #s\n---------|-------|--------|-------|-------|-----------------\n\u001b[32;1mAll files\u001b[0m|\u001b[32;1m    100\u001b[0m|\u001b[32;1m     100\u001b[0m|\u001b[32;1m    100\u001b[0m|\u001b[32;1m    100\u001b[0m|\u001b[33;1m                 \u001b[0m\n\u001b[32;1m index.js\u001b[0m|\u001b[32;1m    100\u001b[0m|\u001b[32;1m     100\u001b[0m|\u001b[32;1m    100\u001b[0m|\u001b[32;1m    100\u001b[0m|\u001b[33;1m                 \u001b[0m\n---------|-------|--------|-------|-------|-----------------\n",
```

Comment:
I've been very busy working on getting everything together for nyc@14 so I haven't had a chance to fully review your changes.  I did just use `console.log()` from node repl to display the before and after of this specific case.  I have a couple of issues:
* I think we should have at least one space between any text and `|`.
* This removes the trailing `|` on the right side of the report.

I realize we do not have a `|` prefix on the left side of the table but I really think having it on the right side of the table improves the way this looks.

As for supporting output of more lines it seems `maxCols` does not work.  I just tried setting it to 160 (my terminal window is wide), no improvement.  It looks like the `Uncovered Line #s` column is fixed width.  I think fixing this is an important first step.  Second step might be to default `maxCols` to the terminal width when `opts.file` is null or `-`.  This way my 212 column wide terminal would have the ability to show a very large list of uncovered lines.

Reply:
> I've been very busy working on getting everything together for nyc@14 so I haven't had a chance to fully review your changes

Ok, don't worry.


> * I think we should have at least one space between any text and `|`.
> * This removes the trailing `|` on the right side of the report.

Yes, it does, but it's the only way to get some extra chars... :-/ I agree output would get a bit uglier, but it's the only way. If not, at most if would only be possible to get some whitespaces from stats columns since all of them have the same chars width while it's not needed.


> I realize we do not have a `|` prefix on the left side of the table but I really think having it on the right side of the table improves the way this looks.

A question of taste :-P In fact, having it in right side and not in left side I find it uglier. Also this way output gets be in a compatible way with Markdown tables, that's a plus.


> As for supporting output of more lines it seems `maxCols` does not work. I just tried setting it to 160 (my terminal window is wide), no improvement. It looks like the `Uncovered Line #s` column is fixed width. I think fixing this is an important first step. Second step might be to default `maxCols` to the terminal width when `opts.file` is null or `-`. This way my 212 column wide terminal would have the ability to show a very large list of uncovered lines.

Good point. I have the same feeling than you that it should default to terminal columns width, but didn't want do change that behaviour because seemed by default it gets unlimited with (at least based how it works for file names), useful for CI output for example, and if so the `maxCols` should be set to the terminal width by the caller executable, or at least that was what code seemed to do.